app/main.py

```diff
@@ -1,8 +1,28 @@
 class Person:
-    # write your code here
-    pass
+    people = {}
+
+    def __init__(self, name: str, age: int) -> None:
+        self.name = name
+        self.age = age
 
 
 def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+    person_list = []
+    for person in people:
+        if person["name"] not in Person.people:
+            person_init = Person(person["name"], person["age"])
+            Person.people[person["name"]] = person_init
+            person_list.append(person_init)
```

Comment:
According to the checklist, when creating a list of instances from a collection, it is recommended to use list comprehension rather than traditional `for` loops. Consider refactoring this part of the code to use list comprehension.